plugins/destination/duckdb/client/transform_gotype.go

```diff
@@ -0,0 +1,159 @@
+package client
+
+import (
+	"database/sql/driver"
+	"reflect"
+	"time"
+
+	"github.com/apache/arrow/go/v15/arrow"
+	"github.com/apache/arrow/go/v15/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v4/types"
+	"github.com/marcboeker/go-duckdb"
+)
+
+func getTypedNilValue(arr arrow.Array) any {
+	switch arr.DataType().(type) {
+	case *types.UUIDType:
+		return nilPtrOf[duckdb.UUID]()
+	case *arrow.TimestampType, *arrow.Date64Type, *arrow.Date32Type:
+		return nilPtrOf[time.Time]()
+	case *arrow.BooleanType:
+		return nilPtrOf[bool]()
+	case *arrow.Int8Type:
+		return nilPtrOf[int8]()
+	case *arrow.Int16Type:
+		return nilPtrOf[int16]()
+	case *arrow.Int32Type:
+		return nilPtrOf[int32]()
+	case *arrow.Int64Type:
+		return nilPtrOf[int64]()
+	case *arrow.Uint8Type:
+		return nilPtrOf[uint32]() // use uint32
+	case *arrow.Uint16Type:
+		return nilPtrOf[uint32]() // use uint32
+	case *arrow.Uint32Type:
+		return nilPtrOf[uint32]()
+	case *arrow.Uint64Type:
+		return nilPtrOf[uint64]()
+	case *arrow.Float32Type:
+		return nilPtrOf[float32]()
+	case *arrow.Float64Type:
+		return nilPtrOf[float64]()
+	case *arrow.StringType:
+		return nilPtrOf[string]()
+	case *arrow.BinaryType, *arrow.LargeBinaryType, *arrow.FixedSizeBinaryType:
+		return nilPtrOf[[]byte]()
+	case *arrow.ListType:
+		arr := arr.(*array.List)
+		v := getTypedNilValue(arr.ListValues())
+		return reflect.New(reflect.SliceOf(reflect.TypeOf(v))).Interface()
+	case *arrow.LargeListType:
+		arr := arr.(*array.LargeList)
+		v := getTypedNilValue(arr.ListValues())
+		return reflect.New(reflect.SliceOf(reflect.TypeOf(v))).Interface()
+	case *arrow.StructType:
+		// Can't create a Go struct dynamically and maps are unsupported: use string
+		return nilPtrOf[string]()
+	case *arrow.MapType:
```

Comment:
should be moved before `arrow.ListLikeType` case

Reply:
Fixed in https://github.com/cloudquery/cloudquery/pull/16668/commits/34abf5cb6234ebd7cbf36ac97b05fe9f2b564723